src/matrix/kaldi-matrix.cc

```diff
@@ -2717,6 +2717,51 @@ void MatrixBase<Real>::AddToRows(Real alpha, Real *const *dst) const {
   }
 }
 
+template<typename Real>
+void MatrixBase<Real>::ParametricRelu(
+                       const MatrixBase<Real> &src,
+                       const VectorBase<Real> &a,
+                       const VectorBase<Real> &b) {
+  KALDI_ASSERT(SameDim(*this, src));
+  MatrixIndexT num_rows = num_rows_, num_cols = num_cols_, stride = stride_, src_stride = src.stride_;
+  KALDI_ASSERT(a.Dim() == num_cols && b.Dim() == num_cols);
+  Real *data = data_;
+  const Real *src_data = src.data_;
+  const Real *adata = a.Data();
+  const Real *bdata = b.Data();
+  for (MatrixIndexT r = 0; r < num_rows; r++) {
+    for (MatrixIndexT c = 0; c < num_cols; c++)
+      data[c] = (src_data[c] > 0 ? adata[c] * src_data[c] : bdata[c] * src_data[c]);
```

Comment:
Hi, it would be more consistent to change: adata -> a_data, bdata -> b_data. The comparison 'src_data[c] > 0' is done with floats, by writing it as 'src_data[c] > 0.0' we can be show it explicitly...